libsolidity/inlineasm/AsmCodeGen.cpp

```diff
@@ -217,42 +228,14 @@ class CodeTransform: public boost::static_visitor<>
 		int height = m_state.assembly.deposit();
```

Comment:
I don't think a syntactic check at a higher level should replace a syntactic check at a lower level because that should be not economical.  The lower level should be close to the machine and therefore should be easier to check.  Otherwise,  the lower level is introducing excess complexity.